packages/source-iottwinmaker/src/utils/propertyValueUtils.ts

```diff
@@ -0,0 +1,62 @@
+import { Type } from '@aws-sdk/client-iottwinmaker';
+import { isEmpty, isNil, isNumber, isString } from 'lodash';
+import { DATA_TYPE } from '@iot-app-kit/core';
+import type { DataValue, DataType as TMDataType } from '@aws-sdk/client-iottwinmaker';
+import type { Primitive, DataType } from '@iot-app-kit/core';
+
+/**
+ * Check if value is not null and not undefined.
+ *
+ * @param value The value to check.
+ * @returns Returns false if the value is nullish, otherwise true.
+ */
+export const isDefined = <T>(value: T | null | undefined): value is T => !isNil(value);
+
+/**
+ * Extracts the value out of a TwinMaker property's DataValue
+ *
+ * NOTE: Currently we treat booleans and objects as strings.
+ */
+export const toValue = (dataValue: DataValue): Primitive | undefined => {
```

Comment:
Maybe we can rename to `propertyDataValue`?

Reply:
What's the reason for extracting only propertyValues?

Reply:
this is a function moved from another file and follow the name with SW source. the purpose of this function is just to get the actual value from the `DataValue` type